test/test_utils/test_sklearn_wrapper.py

```diff
@@ -0,0 +1,112 @@
+                                             random_state=random_seed, verbose=3, n_iter=n_iter)
+        # make a partial for running SMAC under same conditions
+        obj_function = partial(mbo_wrapper._obj_function, base_estimator=classifier, cv_iter=list(StratifiedKFold(3).split(X, y)), X=X, y=y)
+        mbo_wrapper.fit(X, y)
+
+        # Smoke test the score etc:
+        mbo_wrapper.score(X, y)
+        mbo_wrapper.predict_proba(X)
+
+        # Test exception handling on scoring
+        mbo_wrapper.scoring = 'sklearn'
+        assert_raises(ValueError, mbo_wrapper.fit, X, y)
+
+        mbo_params = mbo_wrapper.best_estimator_.get_params()
+
+        # SMAC scenario oject
+        scenario = Scenario({"run_obj": "quality",  "runcount-limit": n_iter, "cs": config_space, "deterministic": "true", "memory_limit": 3072})
+
+        # To optimize, we pass the function to the SMAC-object
+        smac = SMAC(scenario=scenario, rng=random_seed,
+                    tae_runner=obj_function)
+        smac_incumbent = smac.optimize()
+        smac_inc_value = np.mean(np.array(smac.get_tae_runner().run(smac_incumbent, 1)[3]['test_scores']))
+
+        # note that multiple configurations could have led to the same
+        # SMAC always takes the most recently found (in case of 1 instance problems)
+        # so we find the last idx in mbo results
+        mbo_idx = n_iter - 1 - np.argmax(mbo_wrapper.cv_results_['mean_test_score'][::-1])
+
+        self.assertAlmostEqual(mbo_wrapper.best_score_, smac_inc_value, 2) # apparently, scikit-learn search returns only 2 digits
+        for param_name, param_value in smac_incumbent.get_dictionary().items():
+            self.assertIn(param_name, mbo_params)
+            self.assertEqual(mbo_wrapper.cv_results_['param_%s'%param_name][mbo_idx], param_value, msg="param: %s" %param_name)
+
+    def test_mbo_wrapper_dummy(self):
+        iris = datasets.load_iris()
+        X = iris.data
+        y = iris.target
+        classifier = DummyClassifier()
+        cs = ConfigurationSpace()
+        cs.add_hyperparameter(UniformIntegerHyperparameter("random_state", 1, 50, default=1))
+
+        self._compare_with_smac(X, y, classifier, cs, random_seed=42, n_iter=5)
+
+    def test_mbo_wrapper_decision_tree(self):
+        iris = datasets.load_iris()
+        X = iris.data
+        y = iris.target
+
+        classifier = DecisionTreeClassifier()
+        cs = ConfigurationSpace()
+        cs.add_hyperparameter(UniformIntegerHyperparameter('max_leaf_nodes', 4, 32, default=4))
+        cs.add_hyperparameter(UniformIntegerHyperparameter('min_samples_leaf', 1, 128, default=1))
+        cs.add_hyperparameter(UniformIntegerHyperparameter('max_depth', 1, 32, default=1))
+        cs.add_hyperparameter(CategoricalHyperparameter('criterion', ['gini', 'entropy'], default='gini'))
+
+        self._compare_with_smac(X, y, classifier, cs, random_seed=42, n_iter=5)
+
+
```

Comment:
It would be great to have some unit tests, too; for example checking the construction of the SMAC object, the target function, the extraction of results into the CV object etc.